packages/cli/src/oclif/commands/promote.js

```diff
@@ -0,0 +1,153 @@
+const _ = require('lodash');
+const colors = require('colors/safe');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const utils = require('../../utils');
+const { buildFlags } = require('../buildFlags');
+
+const serializeErrors = errors => {
+  const opener = 'Promotion failed for the following reasons:\n\n';
+  if (typeof errors[0] === 'string') {
+    // errors is an array of strings
+    return opener + errors.map(e => `* ${e}`).join('\n');
+  }
+
+  const issues = utils.flattenCheckResult({ errors: errors });
+  return (
+    opener +
+    issues
+      .map(i => `* ${i.method}: ${i.description}\n ${colors.gray(i.link)}`)
+      .join('\n')
+  );
+};
+
+class PromoteCommand extends BaseCommand {
+  async perform() {
+    await utils.checkCredentials();
+
+    const version = this.args.version;
+
+    let shouldContinue;
+    const changelog = await utils.getVersionChangelog(version);
+    if (changelog) {
+      this.log(colors.green(`Changelog found for ${version}`));
+      this.log(`\n---\n${changelog}\n---\n`);
+
+      shouldContinue = await this.confirm(
+        'Would you like to continue promoting with this changelog?'
+      );
+    } else {
+      this.log(
+        `${colors.yellow(
+          'Warning!'
+        )} Changelog not found. Please create a CHANGELOG.md file in a format similar to ${colors.cyan(
+          'https://gist.github.com/xavdid/b9ede3565f1188ce339292acc29612b2'
+        )} with user-facing descriptions.`
+      );
+
+      shouldContinue = await this.confirm(
+        'Would you like to continue promoting without a changelog?'
+      );
+    }
+
+    if (!shouldContinue) {
+      throw new Error('Cancelled promote.');
+    }
+
+    const app = await utils.getLinkedApp();
+    this.log(
+      `Preparing to promote version ${version} of your app "${app.title}".`
+    );
+
+    const body = {};
+    if (changelog) {
+      body.changelog = changelog;
+    }
+
+    this.startSpinner(`Verifying and promoting ${version}`);
+
+    const url = `/apps/${app.id}/versions/${version}/promote/production`;
+    try {
+      await utils.callAPI(
+        url,
+        {
+          method: 'PUT',
+          body
+        },
+        true
+      );
+    } catch (response) {
+      this.stopSpinner();
```

Comment:
maybe move this `stop` farther down so we can stop it with a ✅ or ❌ depending on if it's an actual error or not?

Reply:
Good catch! Fixed in 5abaa9c.